pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/retention/RetentionManagerTest.java

```diff
@@ -197,6 +200,15 @@ public Object answer(InvocationOnMock invocationOnMock)
         return null;
       }
     }).when(resourceManager).deleteSegments(anyString(), ArgumentMatchers.anyList());
+
+    // fake segment lineage.
+    SegmentLineage segmentLineage = new SegmentLineage(REALTIME_TABLE_NAME);
```

Comment:
Why do we need to mock a segment lineage? Is this change related?

Reply:
good catch not anymore since we fake the segment deletion manager. forgot to clean up